core/src/main/python/prepare_model.py

```diff
@@ -241,13 +241,19 @@ def walk(self):
 
         # create a merged model that is not substituted
         merged_model_dictionary = {}
-
         try:
             model_file_list = self.model_files.split(',')
+            target = self.model_context.get_target()
+
             for model_file in model_file_list:
                 if os.path.splitext(model_file)[1].lower() == ".yaml":
                     model_file_name = model_file
-                    FileToPython(model_file_name, True).parse()
+                    
+                if target is not None and os.path.basename(model_file_name) == (target + "-domain.yaml"):
```

Comment:
This should check against the the list of `additional_output` files (not prepended also).

Reply:
fixed